test/e2e/network/networking.go

```diff
@@ -47,14 +47,21 @@ var _ = SIGDescribe("Networking", func() {
 
 	ginkgo.It("should provide Internet connection for containers [Feature:Networking-IPv4]", func() {
 		ginkgo.By("Running container which tries to ping 8.8.8.8")
-		framework.ExpectNoError(
-			framework.CheckConnectivityToHost(f, "", "ping-test", "8.8.8.8", framework.IPv4PingCommand, 30))
+		if !framework.ProviderIs("azure") {
+			framework.ExpectNoError(
+				framework.CheckConnectivityToHost(f, "", "ping-test", "8.8.8.8", framework.IPv4PingCommand, 30))
+		} else {
+			framework.ExpectNoError(
+				framework.CheckConnectivityToHost(f, "", "ping-test", "8.8.8.8", framework.IPv4NcCommand, 30))
```

Comment:
seems complex to have the ProviderIs("azure") logic in two places. Is it possible to make it so that is all encapsulated in CheckConnectivityToHost?

Reply:
This is just for readability. I pass in ```framework.IPv4NcCommand``` here so that ppl will know the actual command being used without checking ```framework.CheckConnectivityToHost```. But I can certainly put them altogether in ```framework.CheckConnectivityToHost```. What do you think?

Reply:
Agreed with @bowei, maybe a function (e.g. connectivtyCommand) and pass the returned slice of strings representing commands into `CheckConnectivityToHost` instead of trying to fit `nc` into `PingCommand`. Thoughts?

Reply:
I encaped all implementation details in ```CheckConnectivityToHost``` now. Thanks for the comments!